code/modules/clothing/glasses/glasses.dm

```diff
@@ -241,6 +241,56 @@
 			hud = null
 			qdel(hud)
 
+/obj/item/clothing/glasses/sunglasses/sechud/syndishades
+	desc = "Strangely ancient technology used to help provide rudimentary eye cover. Enhanced shielding blocks many flashes."
+	name = "sunglasses"
+	icon_state = "sun"
+	item_state = "sunglasses"
+	darkness_view = 0 //Subtly better than normal shades
+	origin_tech = Tc_SYNDICATE + "=3"
+	actions_types = list(/datum/action/item_action/change_appearance_shades)
+	var/list/clothing_choices = list()
+	var/full_access = FALSE
+
+/obj/item/clothing/glasses/sunglasses/sechud/syndishades/New()
+	..()
+	for(var/Type in existing_typesof(/obj/item/clothing/glasses) - /obj/item/clothing/glasses - typesof(/obj/item/clothing/glasses/sunglasses/sechud/syndishades))
+		clothing_choices += new Type
+
+/obj/item/clothing/glasses/sunglasses/sechud/syndishades/attackby(obj/item/I, mob/user)
+	..()
+	if(istype(I, /obj/item/clothing/glasses/sunglasses/sechud) || istype(I, /obj/item/clothing/glasses/hud/security))
+		var/obj/item/clothing/glasses/sunglasses/sechud/syndishades/S = I
+		if(istype(S) && !S.full_access)
+			return
+		if(full_access)
+			to_chat(user, "<span class='warning'>\The [src] already has those access codes.</span>")
+			return
+		else
+			to_chat(user, "<span class='notice'>You transfer the security access codes from \the [I] to \the [src].</span>")
+			full_access = TRUE
+
+/datum/action/item_action/change_appearance_shades
+	name = "Change Shades Appearance"
+
+/datum/action/item_action/change_appearance_shades/Trigger()
+	var/obj/item/clothing/glasses/sunglasses/sechud/syndishades/T = target
+	if(!istype(T))
+		return
+	T.change()
+
+/obj/item/clothing/glasses/sunglasses/sechud/syndishades/proc/change()
+	var/obj/item/clothing/glasses/A
+	A = input("Select style to change it to", "BOOYEA", A) as null|anything in clothing_choices
+	if(!A ||(usr.stat))
+		return
+	desc = A.desc
```

Comment:
Do these even work? Don't you have to use `initial()` here?

Reply:
It works.

Reply:
Oh, of course it does, you're physically creating every item and storing them in a list, FOR EACH GLASS
I suggest you:
- Make the list static, give it a default value of null
- In New(), if the var for the list is null, initialize it like you're doing right now, but append `Type` instead of `new Type`
- use `initial(A.desc)` here and similar below